### ISSUE TITLE
[Feature request]: allow temporarily disabling tombstone compaction

### ISSUE THREAD

**Issue by vladzcloudius** [scylladb/scylladb#13571]
*Installation details*
HEAD: c501163f955e8f665211fb1456f92fcedf07dd3d

*Description*
Add API similar to `nodetool disableautocompaction/enableautocompaction <table name>` that would disable/enable tombstone compactions.

**Comment by raphaelsc** [comment]
> _Installation details_ HEAD: [c501163](https://github.com/scylladb/scylladb/commit/c501163f955e8f665211fb1456f92fcedf07dd3d)
> 
> _Description_ Add API similar to `nodetool disableautocompaction/enableautocompaction <table name>` that would disable/enable tombstone compactions.
> 
> Having this is useful when you want to prevent eviction of expired tombstones for whatever reason: e.g. nodes outage longer than `gc_grace_seconds`.

Less relevant with repair-mode tombstone GC, right?

**Comment by vladzcloudius** [comment]
> > _Installation details_ HEAD: [c501163](https://github.com/scylladb/scylladb/commit/c501163f955e8f665211fb1456f92fcedf07dd3d)
> > _Description_ Add API similar to `nodetool disableautocompaction/enableautocompaction <table name>` that would disable/enable tombstone compactions.
> > Having this is useful when you want to prevent eviction of expired tombstones for whatever reason: e.g. nodes outage longer than `gc_grace_seconds`.
> 
> Less relevant with repair-mode tombstone GC, right?

Wrong. It's very much relevant because conditioning tombstones eviction on a successful repair is not always acceptable.

For example in the following use case this will be completely unacceptable:
* Cluster and a data set are big enough.
* A use case is such that both frequently deletes rows in a multi-row partitions and is read (single rows) latency sensitive.

In such a case relying on a full repair is going to unavoidably make reads suffer from row tombstones accumulated in partitions.

A solution in such a case is to use CL=ALL for deletes and a very short `gc_grace_seconds`.

However one would have to switch to CL="less than all" if one of more nodes go down for whatever reason. During that time you want to disable tombstone compactions in order to avoid eviction of expired tombstones until the cluster is fixed and all deletes that were issued with CL="less than all" are "fixed" (e.g. resent with CL=ALL, or the corresponding data is read with CL=ALL, or the corresponding token is explicitly repaired). After that one can re-enable tombstone compactions.

Today the only way to disable tombstones compactions would be to increase `gc_grace_seconds`. However we all know that doing so when one or more nodes are down is not the best idea.

**Comment by asias** [comment]
We already have:

cqlsh> ALTER TABLE ks.cf WITH tombstone_gc = {'mode':'disabled'};

to disable tombstone gc completely.
